src/systems/filecoin_markets/storage_market/storage_market_actor.go

```diff
@@ -0,0 +1,110 @@
+package storage_market
+
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import deal "github.com/filecoin-project/specs/systems/filecoin_markets/deal"
+import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+
+func (sma *StorageMarketActor_I) WithdrawBalance(balance actor.TokenAmount) {
+	var msgSender addr.Address // TODO replace this from VM runtime
+
+	if balance < 0 {
+		panic("Error")
+	}
+
+	senderBalance, found := sma.Balances()[msgSender]
+	if !found {
+		panic("Error") // TODO replace with fatal
+	}
+
+	if senderBalance.Available() < balance {
+		panic("Error") // TODO replace with fatal
+	}
+
+	senderBalance.Impl().Available_ = senderBalance.Available() - balance
+	sma.Balances()[msgSender] = senderBalance
+
+	// TODO send funds to msgSender with `send` in VM runtime
+}
+
+func (sma *StorageMarketActor_I) AddBalance(balance actor.TokenAmount) {
+	var msgSender addr.Address // TODO replace this
+
+	// TODO subtract balance from msgSender
+	// TODO add balance to StorageMarketActor
+	if balance < 0 {
+		panic("Error")
+	}
+
+	senderBalance, found := sma.Balances()[msgSender]
+	if found {
+		senderBalance.Impl().Available_ = senderBalance.Available() + balance
+		sma.Balances()[msgSender] = senderBalance
+	} else {
+		sma.Balances()[msgSender] = &StorageParticipantBalance_I{
+			Locked_:    0,
+			Available_: balance,
+		}
+	}
+}
+
+func (sma *StorageMarketActor_I) CheckLockedBalance(participantAddr addr.Address) actor.TokenAmount {
+	var msgSender addr.Address // TODO replace this
+
+	return sma.Balances()[msgSender].Locked()
+}
+
+func (sma *StorageMarketActor_I) PublishStorageDeal(newStorageDeals []deal.StorageDeal) []PublishStorageDealResponse {
```

Comment:
should this be PublishStorageDeals?

Reply:
yes or potentially `RegisterStorageDeals`